flytekit/core/type_engine.py

```diff
@@ -1119,13 +1256,15 @@ def _register_default_type_transformers():
     TypeEngine.register(
         SimpleTransformer(
             "none",
-            None,
+            type(None),
             _type_models.LiteralType(simple=_type_models.SimpleType.NONE),
-            lambda x: None,
-            lambda x: None,
-        )
+            lambda x: Literal(scalar=Scalar(none_type=Void())),
+            lambda x: _check_and_convert_void(x)
+        ),
+        [None],
```

Comment:
Can we add a few unit tests to cover this change? I'm particularly interested in demonstrating that the old way of specifying void literals will also be handled correctly (i.e. users will be able to update flytekit independently from flytepropeller).